Source/Task/WaitTimer_stl.cpp

```diff
@@ -0,0 +1,136 @@
+#include "pch.h"
+#include "WaitTimer.h"
+
+using Deadline = std::chrono::high_resolution_clock::time_point;
+
+class WaitTimerImpl
+{
+public:
+
+    ~WaitTimerImpl()
+    {
+        if (m_thread.joinable())
+        {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            m_abortThread = true;
+            m_cv.notify_all();
+        }
+
+        m_thread.join();
+    }
+
+    HRESULT Initialize(_In_opt_ void* context, _In_ WaitTimerCallback* callback)
```

Comment:
we are spawning a thread per timer now, the old code only ever created a single thread

Reply:
Somehow I missed that in the old code.  I will add that back.

Reply:
Ok, I resurrected most of the old code.  I changed "Add" to "Set" because we only allow one value at a time now, and ensured the unit test tested more than one timer value at the same time.